src/v/coproc/tests/utils/supervisor.cc

```diff
@@ -333,17 +320,30 @@ supervisor::process_batch(process_batch_request&& r, rpc::streaming_context&) {
       });
 }
 
-ss::future<empty_response>
+ss::future<absl::node_hash_set<script_id>> supervisor::registered_scripts() {
+    return _coprocessors.map_reduce0(
+      [](script_map_t& smt) {
+          absl::node_hash_set<script_id> all_keys;
+          for (const auto& p : smt) {
+              all_keys.insert(p.first);
+          }
+          return all_keys;
+      },
+      absl::node_hash_set<script_id>(),
+      [](absl::node_hash_set<script_id> set, absl::node_hash_set<script_id> x) {
+          set.merge(std::move(x));
+          return std::move(set);
+      });
+}
+
+ss::future<int64_t>
 supervisor::heartbeat(empty_request&&, rpc::streaming_context&) {
     if (!*_delay_heartbeat.local()) {
-        vlog(coproclog.info, "Replying ping... heartbeat request");
-        return ss::make_ready_future<empty_response>();
+        auto unique_scripts = co_await registered_scripts();
```

Comment:
Do we need to calculate this on each heartbeat? Maybe store this map inside supervisor and change it on enable/disable_coproc?